api/client/service/opts.go

```diff
@@ -253,6 +307,21 @@ func (m *MountOpt) Value() []swarm.Mount {
 	return m.values
 }
 
+func getMountType(fields []string) (swarm.MountType, []string) {
```

Comment:
@dnephin If it helps, I can submit a PR, but I am getting tired of spending time trying to get us to do this correctly. The intent of #24190 was not to change the base syntax. Splitting valid fields and forking the parser has gone way too far.

Before this PR, the grammar was this:

```
csvfield := <defined by csv parser>
keyvalue := key '=' value >> apply-to-mount
key := /[A-Za-z-_]+/
value := /[A-Za-z-_]+/
```

What is the grammar now?